xarray/core/combine.py

```diff
@@ -208,6 +219,8 @@ def _dataset_concat(datasets, dim, data_vars, coords, compat, positions):
     dim, coord = _calc_concat_dim_coord(dim)
     datasets = [as_dataset(ds) for ds in datasets]
     datasets = align(*datasets, join='outer', copy=False, exclude=[dim])
+    # TODO: compute dask coords with a single invocation of dask.compute()
+    datasets = [_load_coords(ds) for ds in datasets]
```

Comment:
We only need this if the argument `coords='different'` is set. Otherwise we don't need to load coords to determine if they are equal or not.